components/renku_data_services/utils/core.py

```diff
@@ -90,3 +91,71 @@ async def transaction_wrapper(self: _WithSessionMaker, *args: _P.args, **kwargs:
             return await f(self, *args, **kwargs)
 
     return transaction_wrapper
+
+
+def _get_url(host: str) -> str:
+    return f"https://{host}/openbis/openbis/rmi-application-server-v3.json"
+
+
+async def get_openbis_session_token(
+    host: str,
+    username: str,
+    password: str,
+    timeout: int = 12,
+) -> str:
+    """Requests an openBIS session token with the user's login credentials."""
+    login = {"method": "login", "params": [username, password], "id": "2", "jsonrpc": "2.0"}
+    async with httpx.AsyncClient(verify=get_ssl_context()) as client:
+        response = await client.post(_get_url(host), json=login, timeout=timeout)
```

Comment:
issue: I think we have to check the `response.status_code` and raise an error if the request fails (see e.g. in `gitlab.py` in `_query_gitlab_graphql`)

Reply:
True. I will add a unit test for it.